### DISCUSSION TITLE
Are the provider links broken on non standard ports?

### DISCUSSION THREAD

**Discussion by programm-ingovals** [nextauthjs/next-auth#3661] [Help] [answered]
Testing out I used the these settings.

```
    IdentityServer4Provider({
      id: "demo",
      clientId: "interactive.confidential",
      clientSecret: "secret",
      issuer: "https://demo.identityserver.io",
    }),
```
If you change the the port number from the default port 3000 the auth process breaks. The link from the page selecting the signin providers sends us here _http://localhost:3000/api/auth/signin/demo_

I assume the default is to overwrite the loginpage but perhaps changes could be made. For example I'm just making a quick proof of concept and wasn't planning on changing.

I haven't looked at the code so I'm not sure if I'm correct with the cause and perhaps this isn't the same for all providers.

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/configuration/options#nextauth_url
